sdk/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltinFun.scala

```diff
@@ -2437,7 +2532,13 @@ private[lf] object SBuiltinFun {
         val needValidationCall: Boolean =
```

Comment:
This condition will need to change once we start propagating disclosed contracts via `ResultNeedContract` to basically always validate on submission as well, am I correct?